packages/jest-worker/src/workers/ChildProcessWorker.js

```diff
@@ -199,17 +103,36 @@ export default class {
           }
         }
 
-        onProcessEnd(error, null);
+        this._onProcessEnd(error, null, this);
         break;
 
       default:
         throw new TypeError('Unexpected response from worker: ' + response[0]);
     }
   }
 
-  _exit(exitCode: number) {
+  onExit(exitCode: number) {
     if (exitCode !== 0) {
-      this._initialize();
+      this.initialize();
     }
   }
+
+  send(request: ChildMessage, onProcessStart: OnStart, onProcessEnd: OnEnd) {
+    onProcessStart(this);
+    this._onProcessEnd = onProcessEnd;
+    // $FlowFixMe
```

Comment:
?

Reply:
¿
`this._child.send(request)` doesn't expect to receive a `request` type